include/llbuild/Core/DependencyKeyIDs.h

```diff
@@ -26,20 +26,49 @@ namespace core {
 /// compact form.
 // FIXME: At some point, figure out the optimal representation for this
 // data structure, which is likely to be a lot of the resident memory size.
-class AttributedKeyIDs {
+class DependencyKeyIDs {
   // Dependencies to be kept track of.
   std::vector<KeyID> keys;
 
-  /// Flags associated with the keys.
-  std::vector<bool> flags;
-
+  /// Flags about the dependency relation, by bit.
+  ///
+  /// value & 1: Flag indicating if the dependency invalidates the downstream task
+  /// (value >> 1) & 1: Flag indicating if the dependency is valid for only the current build and discarded for incremental builds
+  std::vector<uint8_t> flags;
+  
+private:
+  bool singleUse(int index) const {
```

Comment:
These accessors should really be `size_type` (which I think is `size_t` here) not `int`.